include/flang/Semantics/symbol.h

```diff
@@ -82,8 +85,10 @@ class SubprogramDetails {
   MaybeExpr bindName_;
   std::vector<Symbol *> dummyArgs_;  // nullptr -> alternate return indicator
   Symbol *result_{nullptr};
+  Scope *entryScope_{nullptr};  // if ENTRY, points to subprogram's scope
```

Comment:
I am a bit lost with the purpose of entry_scope() vs symbol->scope().
Is this to keep the property that `symbol->scope()->symbol() = identity` or does it have other purposes ?

Reply:
`Symbol::scope()` is used by subprograms, derived types, modules, &c. to link symbols to the scopes that they own.  `ENTRY` point symbols don't have their own scopes, and they are not in the scope of the subprogram in which they appear.  For example,
```
subroutine subr
real x
entry ent
end
```
there's the global scope, which contains the symbol subr` that owns the scope that contains `x`.  The symbol `ent` is also in the global scope -- to its callers, it's no different from any other subroutine -- but it doesn't own a scope.  Instead, it links to the scope of `subr` via its `entryScope_` data member.

Reply:
Makes sens, thanks for explaining.